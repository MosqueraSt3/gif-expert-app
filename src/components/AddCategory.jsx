import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const inputSanitized = inputValue.trim()
        if (inputSanitized.length <= 1) return;
        onAddCategory(inputSanitized);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text"
                placeholder="Search gifs" 
                value={ inputValue }
                onChange= { onInputChange  }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
