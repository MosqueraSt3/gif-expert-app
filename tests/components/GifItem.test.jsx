import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('<GifItem /> component', () => { 

    const title = 'The office';
    const url = 'https://test.com/the-office-image.jpg';

    test('should match with the snapshot', () => { 
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    })

    test('should display the image with URL and ALT', () => { 
        render(<GifItem title={ title } url={ url } />);
        const { src, alt } = screen.getByRole('img');
        expect( alt ).toBe(title);
        expect( src ).toBe(url);
    })

    test('should display the title as a text', () => { 
        render(<GifItem title={ title } url={ url } />);
        expect( screen.getByText(title) ).toBeTruthy();
    })

})