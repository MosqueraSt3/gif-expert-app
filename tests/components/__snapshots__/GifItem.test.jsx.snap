// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> component should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="The office"
      src="https://test.com/the-office-image.jpg"
    />
    <p>
      The office
    </p>
  </div>
</div>
`;
